lib/gollum/app.rb

```diff
@@ -613,9 +613,9 @@ def show_history(wikip)
       end
     end
 
-    def show_page_or_file(fullpath)
+    def show_page_or_file(fullpath, raw)
```

Comment:
Would it be sensible to default `raw` to `false` in this method signature?

Reply:
Based on the way it's implemented, it should always be provided a value. So it's technically okay. But I think there's a benefit to providing a default value and ensuring readers see what the "happy path" (read as: non-raw) looks like.

I wouldn't block merge, but I think that this implementation could be simpler by extracting raw-specific concerns to a separate method:

```
def show_page_or_file(fullpath, raw: false)
  return show_raw_page_or_file(fullpath) if raw

  # The already-committed implementation of `#show_page_or_file`.
end

def show_raw_page_or_file(fullpath)
  # Only the parts of the if statement in this diff where the presence
  # of `raw` matters.
end
```

This results in more lines of code, but less room for error.

Reply:
Added the default. Just for fun, I did your suggested version on a different branch: https://github.com/tstein/gollum/commit/02fbf7ad67e4a792a74af6aed8b6953539318219

I also did it the way you suggested - IMO the duplication necessitates some very small helpers, and overall I think it's not worth the extra volume. lmk what you prefer, it'll only take a moment to swap that commit in for this one. :)

Reply:
I like the idea of extracting a `page_not_found` helper (from the alternative branch), I think that improves the readability of the logic overall. More generally, I'm still on board with @tstein 's logic here and using `show_file` to treat a page as a file when `raw` is true.

Reply:
Works for me. This PR is now for https://github.com/tstein/gollum/commit/02fbf7ad67e4a792a74af6aed8b6953539318219.